dags/analyzer_helper/telegram/extract_raw_data.py

```diff
@@ -0,0 +1,145 @@
+from datetime import datetime
+from typing import Dict, List, Optional
+
+from analyzer_helper.telegram.utils.date_time_format_converter import (
+    DateTimeFormatConverter,
+)
+from github.neo4j_storage.neo4j_connection import Neo4jConnection
+from hivemind_etl_helpers.src.utils.mongo import MongoSingleton
+
+
+class ExtractRawInfo:
+    def __init__(self, forum_endpoint: str, platform_id: str):
+        """
+        Initialize the ExtractRawInfo with the forum endpoint, platform id and set up Neo4j and MongoDB connection.
+        """
+        self.neo4jConnection = Neo4jConnection()
+        self.driver = self.neo4jConnection.connect_neo4j()
+        self.forum_endpoint = forum_endpoint
+        self.converter = DateTimeFormatConverter()
+        self.client = MongoSingleton.get_instance().client
+        self.platform_db = self.client[platform_id]
+        self.rawmemberactivities_collection = self.platform_db["rawmemberactivities"]
+
+    def close(self):
+        """
+        Close the Neo4j connection.
+        """
+        self.driver.close()
+
+    def fetch_message_details(self, created_at: Optional[str] = None, comparison: Optional[str] = None) -> list:
+        """
+        Fetch details of messages from the Telegram chat.
+
+        :param created_at: Optional datetime string to filter messages created after this date.
+        :param comparison: Optional comparison operator, either 'gt' for greater than or 'gte' for greater than or equal to.
+        :return: List of dictionaries containing message details.
+        """
+        if comparison:
+            assert comparison in {"gt", "gte"}, "comparison must be either 'gt' or 'gte'"
+
+        where_clause = "WHERE c.title = $chat_title"
+        if created_at and comparison:
+            print(f"Fetching raw data with created_at: {created_at} and comparison: {comparison}")
+            operator = ">" if comparison == "gt" else ">="
+            where_clause += f" AND m.date {operator} $created_at"
+
+        query = f"""
+        MATCH (c:TGChat)<-[:SENT_IN]-(m:TGMessage)
+        {where_clause}
```

Comment:
Please add the condition `WHERE m.text IS NOT NULL`
We want to process all the text messages

Reply:
Added in the `where_clause`